jvm-runtime/plugin/common/error_detector.go

```diff
@@ -1,28 +1,46 @@
 package common
 
 import (
-	"fmt"
 	"io"
 	"strings"
 
 	"github.com/alecthomas/atomic"
 
 	"github.com/block/ftl/common/builderrors"
 	"github.com/block/ftl/common/slices"
+	"github.com/block/ftl/internal/log"
 )
 
 var _ io.Writer = &errorDetector{}
 
 // errorDetector is a writer that forwards output to stdout, while capturing output until a safe state is reached.
 // When safe state is reached, the output is then parsed for errors.
 type errorDetector struct {
+	logger *log.Logger
 	output string
 	ended  atomic.Value[bool]
 }
 
 func (o *errorDetector) Write(p []byte) (n int, err error) {
 	// Forward output to stdout
-	fmt.Printf("%s", string(p))
+	for _, line := range strings.Split(string(p), "\n") {
+		if len(line) == 0 {
+			continue
+		}
+		if cleanLine, ok := strings.CutPrefix(line, "[ERROR] "); ok {
```

Comment:
Maybe this should have been a regex?

Reply:
We can improve it next... because sometimes I see logs with `<timestamp> [INFO] ...` and othertimes I see `<timestamp> INFO` I think? I'm not sure what is causing which logs and what we should handle